app/assets/javascripts/analytics/ecommerce.js

```diff
@@ -0,0 +1,62 @@
+(function () {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+
+  var Ecommerce = function (config) {
+    this.init = function (element) {
+      var self = this;
+      var ecommerceRows = element.find('[data-ecommerce-row]');
+      var startPosition = parseInt(element.data('ecommerce-start-index'));
+
+      ecommerceRows.each(function(index, ecommerceRow) {
+        var $ecommerceRow = $(ecommerceRow);
+
+        var contentId = $ecommerceRow.attr('data-ecommerce-content-id'),
+          path = $ecommerceRow.attr('data-ecommerce-path');
+
+        self._sendImpression($ecommerceRow, contentId, path, index + startPosition);
+        self._trackProductClick($ecommerceRow, contentId, path, index + startPosition);
+      });
+    }
+
+    this._sendImpression = function (_row, contentId, path, position) {
+      // We only send the id to GA as additional product data is linked when it is uploaded.
+      // This approach is taken to avoid the GA data packet exceeding the 8k limit
+      ga('ec:addImpression', {
```

Comment:
Is this using the globally-available `ga` object? I don't think that's how other tracking is done; I think it all passes through [`GOVUK.Analytics`](https://github.com/alphagov/govuk_frontend_toolkit/blob/master/javascripts/govuk/analytics/analytics.js), which then forwards tracking events both to GA, and on to our own S3-hosted tracking.